libsrc/apple2/cpuspeed.s

```diff
@@ -0,0 +1,40 @@
+;
+; Colin Leroy-Mira <colin@colino.net>, 2024
+;
+; void __fastcall__ apple2_slowdown(void)
+; void __fastcall__ apple2_speedup(void)
+;
+
+        .constructor    initcyareg, 8
+        .export         _apple2_slowdown
+        .export         _apple2_speedup
+        .import         apple2_ostype
+
+        .include        "apple2.inc"
+
+initcyareg:
+        lda     apple2_ostype
+        bpl     :+
+        lda     CYAREG
+        sta     cyareg_orig
+:       rts
+
+_apple2_slowdown:
+        lda     apple2_ostype
+        bpl     :+
+        lda     CYAREG
+        and     #%01111111
+        sta     CYAREG
+:       rts
+
+_apple2_speedup:
+        lda     apple2_ostype
+        bpl     :+
+        lda     cyareg_orig
+        sta     CYAREG
+:       rts
```

Comment:
Given that you restore the original speed, the name "speedup" seems to be a misnomer to me.